packages/psss-server/src/routes/reports/ConfirmCountryWeeklyReportRoute.ts

```diff
@@ -37,12 +51,68 @@ export class ConfirmCountryWeeklyReportRoute extends Route {
     const answers = mapUnconfirmedReportToConfirmedAnswers(report.results[0]);
 
     return this.meditrakConnection?.updateOrCreateSurveyResponse(
-      SURVEY_CODE,
+      CONFIRMED_WEEKLY_SURVEY_CODE,
       countryCode,
       week,
       answers,
     );
   }
+
+  async createAlerts(countryCode: string, week: string) {
+    const report = await this.reportConnection?.fetchReport(
+      CONFIRMED_WEEKLY_REPORT_CODE,
+      [countryCode],
+      [week],
+    );
+
+    if (!report || report.results.length === 0) {
+      throw new RespondingError(
+        `Cannot create alerts: no confirmed weekly data found for ${countryCode} - ${week}`,
+        500,
+      );
+    }
+
+    const [result] = report.results;
+    const response: any = {
+      createdAlerts: [],
+    };
+    const previousAlertResponses = await this.meditrakConnection?.findSurveyResponses(
+      ALERT_SURVEY_CODE,
+      countryCode,
+      week,
+    );
+
+    for (const syndromeCode of SYNDROME_CODES) {
+      if (result[`${syndromeCode} Threshold Crossed`] === true) {
+        const surveyResponseId = generateId();
+        await this.meditrakConnection?.createSurveyResponse(
+          ALERT_SURVEY_CODE,
+          countryCode,
+          week,
+          {
+            PSSS_Alert_Syndrome: syndromeCode,
+            PSSS_Alert_Archived: 'No',
+          },
+          surveyResponseId,
+        );
+
+        response.createdAlerts.push({
+          id: surveyResponseId,
+          title: syndromeCode,
+        });
+      }
+    }
+
+    if (previousAlertResponses && previousAlertResponses.length) {
+      // delete the old alerts in case the weekly data is
+      // re-confirmed and there have been some alerts created before
+      for (const alertResponse of previousAlertResponses) {
+        await this.meditrakConnection?.deleteSurveyResponse(alertResponse);
+      }
+    }
```

Comment:
Not sure if this is the best way..

For Alert survey responses, they are no longer unique by the combo of surveyCode/orgUnit/period like weekly reports. The unique combo for the alert survey responses is should be surveyCode/syndrome/orgUnit/period. So in the scenario of creating alerts when **reconfirming data**, to deal with old alerts, the easiest way I've found is to delete them and recreate new ones again.